lib/addons/library.kodi.audioengine/libKODI_audioengine.cpp

```diff
@@ -64,14 +64,17 @@ DLLEXPORT void AudioEngine_unregister_me(void *hdl, void* cb)
 // ---------------------------------------------
 DLLEXPORT CAddonAEStream* AudioEngine_make_stream(void *hdl, void *cb, AEDataFormat DataFormat, unsigned int SampleRate, unsigned int EncodedSampleRate, enum AEChannel *Channels, unsigned int Options)
 {
-  AEStreamHandle *streamHandle = ((CB_AudioEngineLib*)cb)->MakeStream(DataFormat, SampleRate, EncodedSampleRate, Channels, Options);
-  if (!hdl || !cb || !streamHandle)
+  if (cb && hdl)
   {
-    fprintf(stderr, "%s-ERROR: AudioEngine_make_stream can't MakeStrean failed!\n", LIBRARY_NAME);
-    return NULL;
+    AEStreamHandle *streamHandle = ((CB_AudioEngineLib*)cb)->MakeStream(DataFormat, SampleRate, EncodedSampleRate, Channels, Options);
+    if (streamHandle)
+    {
+      return new CAddonAEStream(hdl, cb, streamHandle);
+    }
   }
 
-  return new CAddonAEStream(hdl, cb, streamHandle);
+  fprintf(stderr, "%s-ERROR: AudioEngine_make_stream can't MakeStrean failed!\n", LIBRARY_NAME);
```

Comment:
when touching this can we change it to a proper log call? this prints to /dev/null on windows